src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -334,14 +335,24 @@ public override bool IsBindingRequired
         {
             get
             {
-                if (MetadataKind == ModelMetadataKind.Property)
-                {
-                    return BindingMetadata.IsBindingRequired;
-                }
-                else
+                if (!_isBindingRequired.HasValue)
                 {
-                    return false;
+                    if (MetadataKind == ModelMetadataKind.Type)
+                    {
+                        _isBindingRequired = false;
```

Comment:
The set which special case the metadata kind is `IsBindingAllowed`, `IsBindingRequired` and `IsReadOnly`.

I agree that one reading of the metadata on a `Type` would be "all properties / parameters of this `Type` have XYZ" but that's not how the rest of the system reads the information. So my question is more like "Could customers want to have this mean something on a `Type`?"